### PR TITLE
CORTX-33581 : Automate Alex data in health report

### PR DESCRIPTION
**Pull request by subhalaxmisa**
Signed-off-by: subhalaxmisa <subhalaxmi.sahoo@seagate.com>

# Problem Statement
- Automate Alex data in health report

# Design
-  Automate Alex data in health report

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [x] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM
- [x] Jenkins pipelines used for testing

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [x] Interface change (if any) are documented
- [x] Side effects on other features (deployment/upgrade)
- [x] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [x] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: scripts/automation/alex/alex_report.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+API_URL = "https://github.com/Seagate/"
+response  = json.loads(requests.get("http://cortx-storage.colo.seagate.com/releases/opensource_builds/alex_report/config.json").text)
+global_words = response["global_scope"]
+repo_name=[]
+
+os.system("yum install epel-release -y; curl -fsSL https://rpm.nodesource.com/setup_16.x | sudo -E bash - ; yum install nodejs -y; npm install alex --global -y")
+os.system("yum install python3 -y; python3 -m pip install --upgrade pip; pip install virtualenv; virtualenv alex --python=python3.6, source ./alex/bin/activate; pip install GitPython==2.1.15, NODE_OPTIONS='--max-old-space-size=1536' ")
+
+for i in response["repository"]:
```
[the remark this answers is not in the document]
Comment (subhalaxmisa) [reply scripts/automation/alex/alex_report.py]: Done

**Review by gauravchaudhari02 — DISMISSED** [review]

**Review by shailesh-vaidya — COMMENTED** [review]
@subhalaxmisa  - Can you please add the Jenkins job used for testing these changes.

File: scripts/automation/alex/alex_pickle.py [hunk 2]
```diff
@@ -74,4 +74,4 @@
     print ("Updated the pickle file successfully:\n %s" %pickle.load(f))
 
 
-#print (alex_report_status)
+#print (alex_report_status)
```
Comment (shailesh-vaidya) [inline scripts/automation/alex/alex_pickle.py:77]: I guess there is no change in this file. Can we remove this from PR

File: scripts/reports/repo-health.sh [hunk 3]
```diff
@@ -31,6 +31,14 @@ yum install gcc python3-devel git-lfs pango-devel -y
 pip3 install seaborn matplotlib pandas xlrd requests python-dateutil pyGithub jupyterlab django-weasyprint weasyprint==52.5
 git lfs install
 
+#Install alex
+yum install epel-release -y; curl https://raw.githubusercontent.com/creationix/nvm/v0.33.11/install.sh | bash
+source ~/.bashrc
+nvm --version
+nvm install 14.20.0
+npm install alex --global -y
+alex --version
```
Comment (shailesh-vaidya) [inline scripts/reports/repo-health.sh:40]: Please add some echo statements so it will be easy to search in the logs

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]

File: scripts/reports/repo-health.sh (hunk 3 above)
Comment (subhalaxmisa) [reply scripts/reports/repo-health.sh:40]: Okay

File: scripts/automation/alex/alex_pickle.py (hunk 2 above)
Comment (subhalaxmisa) [reply scripts/automation/alex/alex_pickle.py]: Yes and I removed it.

File: scripts/reports/repo-health.sh (hunk 3 above)
Comment (subhalaxmisa) [reply scripts/reports/repo-health.sh:40]: Let me know what more echo statements are required.

Comment (shailesh-vaidya) [reply scripts/reports/repo-health.sh:40]: You can fetch and print "Node Version: $NODE_VERSION  Alex Version: $ALEX_VERSION"

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]

**Review by shailesh-vaidya — APPROVED** [review]

**Comment by subhalaxmisa** [comment]
Job : (https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/Reports/job/repo-health/17/console)

File: scripts/reports/repo-health.sh [hunk 4]
```diff
@@ -31,6 +31,16 @@ yum install gcc python3-devel git-lfs pango-devel -y
...
+alex_version=$(alex -v 2>&1); echo $alex_version | grep -o '[0-9.]*'
```
Comment (gauravchaudhari02) [inline scripts/reports/repo-health.sh:41]: please add string `Alex Version: ` in echo as well while printing version

File: scripts/reports/repo-health.sh [hunk 5]
```diff
@@ -31,6 +31,16 @@ yum install gcc python3-devel git-lfs pango-devel -y
...
+node_version=$(node -v 2>&1); echo $node_version | grep -o '[0-9.]*'
```
Comment (gauravchaudhari02) [inline scripts/reports/repo-health.sh:42]: please add string `Node Version: ` in echo as well while printing version

**Comment by gauravchaudhari02** [comment]
Please fix CLA check

File: scripts/reports/repo-health.sh (hunk 5 above)
Comment (subhalaxmisa) [reply scripts/reports/repo-health.sh]: Done

File: scripts/reports/repo-health.sh (hunk 4 above)
Comment (subhalaxmisa) [reply scripts/reports/repo-health.sh]: Done

**Review by gauravchaudhari02 — APPROVED** [review]

**Review by shailesh-vaidya — APPROVED** [review]

**Pull request merged**